tt_metal/impl/dispatch/device_command.cpp

```diff
@@ -118,73 +140,107 @@ void DeviceCommand::add_dispatch_write_paged(bool flush_prefetch, uint8_t is_dra
     uint32_t payload_sizeB = sizeof(CQDispatchCmd) + data_sizeB;
     this->add_prefetch_relay_inline(flush_prefetch, payload_sizeB);
 
-    CQDispatchCmd write_cmd;
+    uint32_t size_written_to_cmd_regionB = sizeof(CQDispatchCmd);
+    if (data != nullptr) {
```

Comment:
minor: looks like this code is re-used for both inline and non-inline data.  for non-inline data the command is small and the overhead of the test may matter. for inline data, probably not.  but this could be templated, right?  then the tests will get optimized out

Reply:
agreed, will uplift. Paused on additional clean up here to debug my functionality regression